sycl/source/detail/platform_impl.cpp

```diff
@@ -247,47 +247,40 @@ static void filterAllowList(vector_class<RT::PiDevice> &PiDevices,
         if (!std::regex_match(PlatformName, std::regex(Desc.PlatName))) {
           PlatNameState = MatchState::NOMATCH;
           continue;
-        } else {
-          PlatNameState = MatchState::MATCH;
         }
       }
 
       if (!Desc.PlatVer.empty()) {
         if (!std::regex_match(PlatformVer, std::regex(Desc.PlatVer))) {
           PlatVerState = MatchState::NOMATCH;
           continue;
-        } else {
-          PlatVerState = MatchState::MATCH;
         }
       }
 
       if (!Desc.DevName.empty()) {
         if (!std::regex_match(DeviceName, std::regex(Desc.DevName))) {
           DevNameState = MatchState::NOMATCH;
           continue;
-        } else {
-          DevNameState = MatchState::MATCH;
         }
       }
 
       if (!Desc.DevDriverVer.empty()) {
         if (!std::regex_match(DeviceDriverVer, std::regex(Desc.DevDriverVer))) {
           DevVerState = MatchState::NOMATCH;
           continue;
-        } else {
-          DevVerState = MatchState::MATCH;
```

Comment:
We have three states of matching: matched, not matched, unknown (unchecked). With these changes only two states are employed now: not matched and unknown.
It looks like we have to think over the logic of matching and states of matching.

Reply:
Agree, removed matched state as unnecessary now, didn't change logic here yet as waiting for LIT test results (as prev results failed). I do not like the current enum either - only UNKNOWN and NOMATCH look strange. Will think about proper refactoring. Do you have any ideas?

Reply:
Well, as you said, having unknown and not matched states is quite a ridiculous thing. I believe, it should be matched and not matched with default values properly set. As far as I see from current changes it's going to be allow/deny logic: allow by default, deny when checked.
Hence, we have to define which device should and should not pass the allow-list filter. As far as I understand only a device of the same platform name/version and device name/version should pass.

Reply:
Thanks! I replaced it with allow/deny logic.